commit f8d7e13338618d664cb171227ba870791344e95b
Author: nephros <nemo@pgxperiiia10>
Date:   Tue Apr 28 09:07:50 2026 +0200

    [Sailfish OS] cmake: respect no-color environment
    
    Coloured output makes OBS build logs unreadable

diff --git a/cmake/sdlcompilers.cmake b/cmake/sdlcompilers.cmake
index cd4dde0f6..13639af6e 100644
--- a/cmake/sdlcompilers.cmake
+++ b/cmake/sdlcompilers.cmake
@@ -151,15 +151,17 @@ function(SDL_AddCommonCompilerFlags TARGET)
     endif()
   endif()
 
-  if(USE_CLANG)
-    check_c_compiler_flag("-fcolor-diagnostics" COMPILER_SUPPORTS_FCOLOR_DIAGNOSTICS)
-    if(COMPILER_SUPPORTS_FCOLOR_DIAGNOSTICS)
-      sdl_target_compile_option_all_languages(${TARGET} "-fcolor-diagnostics")
-    endif()
-  else()
-    check_c_compiler_flag("-fdiagnostics-color=always" COMPILER_SUPPORTS_FDIAGNOSTICS_COLOR_ALWAYS)
-    if(COMPILER_SUPPORTS_FDIAGNOSTICS_COLOR_ALWAYS)
-      sdl_target_compile_option_all_languages(${TARGET} "-fdiagnostics-color=always")
+  if(CMAKE_COLOR_DIAGNOSTICS)
+    if(USE_CLANG)
+      check_c_compiler_flag("-fcolor-diagnostics" COMPILER_SUPPORTS_FCOLOR_DIAGNOSTICS)
+      if(COMPILER_SUPPORTS_FCOLOR_DIAGNOSTICS)
+        sdl_target_compile_option_all_languages(${TARGET} "-fcolor-diagnostics")
+      endif()
+    else()
+      check_c_compiler_flag("-fdiagnostics-color=always" COMPILER_SUPPORTS_FDIAGNOSTICS_COLOR_ALWAYS)
+      if(COMPILER_SUPPORTS_FDIAGNOSTICS_COLOR_ALWAYS)
+        sdl_target_compile_option_all_languages(${TARGET} "-fdiagnostics-color=always")
+      endif()
     endif()
   endif()
 
